flink-connectors/flink-connector-hbase/src/test/java/org/apache/flink/connector/hbase/HBaseConnectorITCase.java

```diff
@@ -455,99 +392,82 @@ public void testTableSink() throws Exception {
 
 	@Test
 	public void testTableSourceSinkWithDDL() throws Exception {
+		// only test TIMESTAMP/DATE/TIME/DECIMAL for new connector(using blink-planner), because new connector encodes
+		// DATE/TIME to int, the old one encodes to long, and DECIMAL with precision works well in new connector.
+		final boolean testTimeAndDecimalTypes = BLINK_PLANNER.equals(planner) && !isLegacyConnector;
```

Comment:
This test code is really hard to maintain and read. I would suggest only test for blink planner and new connector for simplification.